flake.nix

```diff
@@ -0,0 +1,184 @@
+          imageDigest = e2e-kind-node-digest;
+          sha256 = e2e-kind-node-sha256;
+          finalImageTag = "v${e2e-kind-node-version}";
+          finalImageName = "kindest/node";
+        };
+
+        cert-manager-controller-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-controller";
+          imageDigest = e2e-cert-manager-controller-digest;
+          sha256 = e2e-cert-manager-controller-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-controller";
+        };
+
+        cert-manager-webhook-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-webhook";
+          imageDigest = e2e-cert-manager-webhook-digest;
+          sha256 = e2e-cert-manager-webhook-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-webhook";
+        };
+
+        cert-manager-cainjector-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-cainjector";
+          imageDigest = e2e-cert-manager-cainjector-digest;
+          sha256 = e2e-cert-manager-cainjector-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-cainjector";
+        };
+
+        cert-manager-ctl-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/jetstack/cert-manager-ctl";
+          imageDigest = e2e-cert-manager-ctl-digest;
+          sha256 = e2e-cert-manager-ctl-sha256;
+          finalImageTag = e2e-cert-manager-version;
+          finalImageName = "quay.io/jetstack/cert-manager-ctl";
+        };
+
+        csi-node-driver-registrar-image = pkgs.dockerTools.pullImage{
+          imageName = "quay.io/k8scsi/csi-node-driver-registrar";
+          imageDigest = csi-node-driver-registrar-digest;
+          sha256 = csi-node-driver-registrar-sha256;
+          finalImageTag = "v${csi-node-driver-registrar-version}";
+          finalImageName = "quay.io/k8scsi/csi-node-driver-registrar";
+        };
+
+        busybox-image = pkgs.dockerTools.pullImage{
+          imageName = "busybox";
+          imageDigest = e2e-busybox-digest;
+          sha256 = e2e-busybox-sha256;
+          finalImageTag = "latest";
+          finalImageName = "busybox";
+        };
+
+      in {
+        packages = {
+          default = containerImage;
+          container = containerImage;
+          inherit cert-manager-helm-chart;
+          inherit kind-node-image;
+          inherit cert-manager-controller-image;
+          inherit cert-manager-webhook-image;
+          inherit cert-manager-cainjector-image;
+          inherit cert-manager-ctl-image;
+          inherit cert-manager-csi;
+          inherit busybox-image;
+          inherit csi-node-driver-registrar-image;
+          inherit csi-lib-e2e;
+        };
+        devShells.default = pkgs.mkShell {
```

Comment:
What's a devShell? 😄 looks like it is provisioning a local development environment? Can I run this on my local machine? How does it work across macOS vs Linux etc?

Reply:
Correct, this is setting up your local environment in your shell. You can access that shell with `nix develop`. 

This should work on all systems, so long as they are included in ` flake-utils.lib.eachDefaultSystem `.

```
$ flake show
git+file:///keep/home/go/src/github.com/cert-manager/csi-lib?ref=refs%2fheads%2fe2e-nix-boilerplate&rev=17ae38eb12de438350b90fdc8e79e15f9362090f
├───devShells
│   ├───aarch64-darwin
│   │   └───default: development environment 'nix-shell'
│   ├───aarch64-linux
│   │   └───default: development environment 'nix-shell'
│   ├───i686-linux
│   │   └───default: development environment 'nix-shell'
│   ├───x86_64-darwin
│   │   └───default: development environment 'nix-shell'
│   └───x86_64-linux
│       └───default: development environment 'nix-shell'
```